### PR TITLE
Add magic getter to retrieve API and improve code completion

### PR DESCRIPTION
**Pull request by airmoi**
Hi,

This commit aim to improve code completion by adding a magic getter function to retrieve Apis.
It preserves backward compatibility.

I've also updated readme samples

### CODE REVIEW

File: lib/Redmine/Client.php [hunk 1]
```diff
@@ -113,6 +155,19 @@ public function __construct($url, $apikeyOrUsername, $pass = null)
         $this->apikeyOrUsername = $apikeyOrUsername;
         $this->pass = $pass;
     }
+    
+    /**
+     * PHP setter magic method.
```
Comment (Art4) [inline lib/Redmine/Client.php]: Setter? Must be 'getter'

Comment (airmoi) [inline lib/Redmine/Client.php]: Your right !

File: lib/Redmine/Client.php [hunk 2]
```diff
@@ -97,6 +117,28 @@ class Client
         JSON_ERROR_CTRL_CHAR => 'Control character error, possibly incorrectly encoded',
         JSON_ERROR_SYNTAX => 'Syntax error',
     );
+    
+    private $classes = array(
+            'attachment' => 'Attachment',
```
Comment (kbsali) [inline lib/Redmine/Client.php]: could you fix the indentation level?

File: lib/Redmine/Client.php [hunk 3]
```diff
@@ -113,6 +155,19 @@ public function __construct($url, $apikeyOrUsername, $pass = null)
...
+    
+    /**
+     * PHP getter magic method.
+     * 
+     * @param string $name
+     * @return mixed property value
+     */
+    public function __get($name)
+    {
+        if(isset($this->classes[$name])){
```
Comment (kbsali) [inline lib/Redmine/Client.php]: cs fix

**Comment by kbsali** [comment]
@airmoi  thanks for this nice and simple PR!
Could you address the 2 code style issues and maybe also update the example.php?

Thanks!

**Comment by airmoi** [comment]
Ok I fixed it and also add a little change to the magic getter so it acts as api() method (throws exception when wrong API is called)

**Comment by kbsali** [comment]
awesome, it would have been even better with some tests! ;)
:+1:

**Pull request merged**